_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
 .infrastructure-live/stage/_global/account-baseline/terragrunt.hcl
 [source,hcl]
 ----
+locals {
+  aws_region = "us-east-1"
+
+  accounts = {
+    logs     = "409800740445"
+    security = "673123100581"
+    shared   = "384759303421"
+    dev      = "293847503945"
+    stage    = "384924092834"
+    prod     = "784260063686"
+    root     = "216044045972"
+  }
+
+  # Both buckets are created in the logs account by account-baseline-root
+  config_s3_bucket_name     = "acme-config-bucket-logs"
+  cloudtrail_s3_bucket_name = "acme-cloudtrail-logs"
+
+  # The Cloudtrail KMS Key is deployed at the logs account but it's value is an output from the root account.
+  cloudtrail_kms_key_arn = "arn:aws:kms:us-east-1:409800740445:alias/cloudtrail-acme"
+
+  # A local for convenient access to the security account root ARN.
+  security_account_root_arn = "arn:aws:iam::${local.accounts.security}:root"
+}
+
 inputs = {
-  # Fill in your region you want to use (only used for API calls) and the ID of your security AWS account
-  aws_region     = "us-east-2"
-  aws_account_id = "<STAGE_ACCOUNT_ID>"
+  # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = local.aws_region
+  aws_account_id = local.accounts["stage"]
 
   # Prefix all resources with this name
-  name_prefix = "<COMPANY_NAME>-stage"
+  name_prefix = "stage-logs"
```

Comment:
Hm, this one I'm not sure about. Setting the `name_prefix` for the environment + account is not what we're currently doing in the ref arch deploys. See the [LZ guide](https://gruntwork.io/guides/foundations/how-to-configure-production-grade-aws-account-structure/#prepare-your-infrastructure-live-repository) for how we set `name_prefix` (in `common.hcl`, usually as the company name, and then in the root `terragrunt.hcl`, so that no downstream `terragrunt.hcl`s need to set it). It makes me wonder if our ref arch is setting the `name_prefix` a little too generic? Pinging @bwhaley for another opinion.

Reply:
It does need to be something unique as it will ultimately be used to create the S3 bucket, which is a global namespace among all AWS users. Identifying a good rule of thumb is tricky because it must be both (a) short enough to not run in to length constraints, and (b) globally unique. `<COMPANY_NAME>` is usually unique enough, but runs the risk of being too long. Many customers have chosen a prefix that is an acronym or shortened identifiable version of the company name, like `phc` or `mf`. 

I think we ought to share this context in the comment on the preceding line, and go with something like:

```
name_prefix = "<SOME UNIQUE IDENTIFIER>-logs"
```

Reply:
Thanks Ben. Seems like we need to update the LZ guide and the Ref Arch code, too. Looking at the `terraform-aws-architecture-catalog`, we set `name_prefix` is only [one place](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/common.hcl#L13-L14), which gets passed around from the [root terragrunt](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/terragrunt.hcl#L25) (and actually it's [duplicated here](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/terragrunt.hcl#L73), which we can simplify). That means it's not unique for the S3 buckets, right?

Reply:
I feel like I must be missing something. It can't be that we use the same `name_prefix` everywhere. Maybe it does get overridden in each account's terragrunt, but I'm not seeing it.

Reply:
@bwhaley Thoughts on this?